internal/host/host_test.go

```diff
@@ -1323,6 +1323,72 @@ var _ = Describe("UpdateInventory", func() {
 			Expect(h.InstallationDiskID).To(Equal(diskId))
 		})
 	})
+	Context("Interfaces", func() {
+		const (
+			diskName = "FirstDisk"
+			diskId   = "/dev/disk/by-id/FirstDisk"
+		)
+		var (
+			physicalInterface = models.Interface{
+				Name: "eth0",
+				IPV4Addresses: []string{
+					"1.2.3.4/24",
+				},
+				IPV6Addresses: []string{
+					"1001:db8::10/120",
+				},
+				Type: "physical",
+			}
+			virtualInterface = models.Interface{
+				Name: "cni1",
+				IPV4Addresses: []string{
+					"10.88.0.0/24",
+				},
+				IPV6Addresses: []string{
+					"1002:db8::10/120",
+				},
+				Type: "device",
+			}
+			dummy = &leader.DummyElector{}
+		)
+
+		BeforeEach(func() {
+			host = hostutil.GenerateTestHost(hostId, infraEnvId, "", models.HostStatusDiscovering)
+			host.Inventory = common.GenerateTestInventoryWithVirtualInterface()
+			host.InstallationDiskPath = ""
+			Expect(db.Create(&host).Error).ShouldNot(HaveOccurred())
+			mockValidator.EXPECT().DiskIsEligible(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any())
+			mockValidator.EXPECT().ListEligibleDisks(gomock.Any()).Return(
+				[]*models.Disk{{ID: diskId, Name: diskName}},
+			)
+		})
+		It("Doesn't save virtual interfaces when virtual interface flag is not enabled", func() {
+			Expect(hapi.UpdateInventory(ctx, &host, host.Inventory)).ToNot(HaveOccurred())
+
+			h := hostutil.GetHostFromDB(hostId, infraEnvId, db)
+			Expect(h.Inventory).ToNot(BeEmpty())
+			inventory, err := common.UnmarshalInventory(h.Inventory)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(inventory).To(Not(BeNil()))
+			Expect(inventory.Interfaces).Should(HaveLen(1))
+			Expect(inventory.Interfaces[0].Type).To(Equal(physicalInterface.Type))
+		})
+		It("Saves virtual interfaces when virtual interface flag is enabled", func() {
+			defaultConfig.EnableVirtualInterfaces = true
+			enabled_hapi := NewManager(common.GetTestLog(), db, mockEvents, mockValidator,
+				nil, createValidatorCfg(), nil, defaultConfig, dummy, nil, nil)
+			Expect(enabled_hapi.UpdateInventory(ctx, &host, host.Inventory)).ToNot(HaveOccurred())
```

Comment:
nit, but when I see `ToNot(HaveOccurred())` I always think we're expecting a method not to be called.

Personally I prefer `Expect(enabled_hapi.UpdateInventory(ctx, &host, host.Inventory)).To(Succeed())` in cases where we're asserting on the error response of a function that returns just an error value.

Reply:
Gotcha, modified it, thank you!